models/RwdImage.php

```diff
@@ -464,6 +478,108 @@ public function get_set_sources() {
 		return $sources;
 	}
 
+	/**
+	 * Dynamically resize image.
+	 *
+	 * @param int    $attach_id Attachment ID.
+	 * @param array  $meta_data Attachment meta data.
+	 * @param string $key       Image size key.
+	 * @param int    $width     Image width.
+	 * @param int    $height    Image height.
+	 * @param int    $crop      Crop image.
+	 *
+	 * @return array
+	 */
+	public function resize_image( $attach_id, $meta_data, $key, $width, $height, $crop ) {
+		if ( ! empty( $meta_data ) && ! isset( $meta_data['sizes'][ $key ] ) && $meta_data['width'] <= $width ) {
+			// for usual images for lower image sizes we will use max image size for the bigger sizes.
+			$meta_data['sizes'][ $key ] = array(
+				'width'      => $meta_data['width'],
+				'height'     => $meta_data['height'],
+				'file'       => basename( $meta_data['file'] ),
+				'rwd_width'  => $width,
+				'rwd_height' => $height,
+				'crop'       => $crop,
+			);
+			// save to cache.
+			$this->set_attachment_metadata( $attach_id, $meta_data );
+			// update metadata.
+			wp_update_attachment_metadata( $attach_id, $meta_data );
+
+			return $meta_data;
+		}
+		// set RWD size config
+		if ( ! isset( $meta_data['sizes'][ $key ] ) ) {
+			$meta_data['sizes'][ $key ] = array(
+				'rwd_width'  => $width,
+				'rwd_height' => $height,
+				'crop'       => $crop,
+			);
+		} elseif ( ! isset( $meta_data['sizes'][ $key ]['rwd_width'] ) ) {
+			$meta_data['sizes'][ $key ] = array(
+				'rwd_width'  => $width,
+				'rwd_height' => $height,
+				'crop'       => $crop,
+			);
+		}
+		$upload_dir    = wp_get_upload_dir();
+		$image_baseurl = $upload_dir['basedir'] . DIRECTORY_SEPARATOR . image_get_intermediate_size( $attach_id, $key )['path'];
+		if ( $meta_data['sizes'][ $key ]['rwd_width'] !== $width || $meta_data['sizes'][ $key ]['rwd_height'] !== $height
+		     || $meta_data['sizes'][ $key ]['crop'] !== $crop || ! file_exists( $image_baseurl )
```

Comment:
crop compare is wrong. crop can be an array (when you specify crop position), so you can't compare arrays with !==. you should compare serialized version with strcmp.

to optimize this check - place file exists as the first condition (because it's the most possible scenario)